src/main/java/com/limechain/runtime/WasmSectionUtils.java

```diff
@@ -0,0 +1,124 @@
+package com.limechain.runtime;
+
+import com.limechain.exception.WasmRuntimeException;
+import com.limechain.runtime.version.scale.RuntimeVersionReader;
+import com.limechain.runtime.version.ApiVersions;
+import com.limechain.runtime.version.RuntimeVersion;
+import com.limechain.utils.ByteArrayUtils;
+import io.emeraldpay.polkaj.scale.ScaleCodecReader;
+import lombok.experimental.UtilityClass;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+// TODO: [Testability concern] Maybe make this class non-static?
+@UtilityClass
+public class WasmSectionUtils {
+    private static final byte[] RUNTIME_VERSION_SECTION_NAME = "runtime_version".getBytes();
+    private static final byte[] RUNTIME_APIS_SECTION_NAME = "runtime_apis".getBytes();
+
+    /**
+     * Parses the runtime version if both wasm custom sections ("runtime_apis" and "runtime_version") are present.
+     * @param wasmBinary the wasm blob
+     * @return the parsed RuntimeVersion if both custom sections are present, null otherwise
+     */
+    @Nullable
+    public RuntimeVersion parseRuntimeVersionFromCustomSections(byte[] wasmBinary) {
+        // byte value of \0asm concatenated with 0x1, 0x0, 0x0, 0x0 from smoldot runtime_version.rs#97
```

Comment:
Not a good practice to point to a concrete line in another repo since it can easily be changed making this comment false.

Reply:
I agree, but I left this comment as it was since I was unable to find anything regarding that prefix in the spec / anywhere in a public source... I'm most probably missing something, but alternatively, we could link to a certain commit hash in the smoldot repo in order to not lose the motivation of this "magic prefix" 😄

Reply:
commit hash is a bit better for sure